src/lib/components/CopyCode/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Element that renders copyable code.
+ */
+
+import {BaseElement} from "../BaseElement";
+
+/**
+ * Renders code block that can easily be copied.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class CopyCode extends BaseElement {
+  static get properties() {
+    return {
+      code: {type: String},
+    };
+  }
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute("tabindex", "0");
+
+    if (this.firstChild) {
+      const copyButton = document.createElement("input");
+      copyButton.setAttribute("type", "button");
+      copyButton.setAttribute("tabindex", "0");
+      copyButton.className = "w-copy-code-button";
+
+      this.firstChild.prepend(copyButton);
+
+      this.addEventListener("click", (e) => {
+        if (e.target === copyButton) {
+          e.preventDefault();
+          this.copyCode();
+        }
+      });
+    }
+  }
+
+  copyCode() {
```

Comment:
This might be a good opportunity to use the newer [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) and fallback to `execCommand` if the browser doesn't support it.

```
if ('clipboard' in navigator) {
  navigator.clipboard.writeText(this.innerText)
} else {
  // fallback
}
```

I think the MDN docs say the Clipboard API requires permissions but in my testing in Chrome and Firefox it didn't prompt for permission. It is worth noting that Safari doesn't support the Clipboard API so we'd need to use a fallback for it. My hope is that eventually they'll ship support and we can remove the fallback.

Btw, I think we might be able to get away with using `innerText` instead of relying on the attribute.